src/controlplane/client/connector.go

```diff
@@ -43,6 +42,22 @@ type defaultConnector struct {
 
 // DefaultConnector returns a defaultConnector that probes all endpoints in the list and return the first responding status OK.
 func DefaultConnector(endpoints []config.Endpoint, kc kubernetes.Interface, inClusterConfig *rest.Config, logger log.Logger) (Connector, error) {
+	if inClusterConfig == nil {
+		return nil, fmt.Errorf("inClusterConfig cannot be nil")
+	}
+
+	if logger == nil {
+		return nil, fmt.Errorf("logger cannot be nil")
+	}
+
+	if kc == nil {
+		return nil, fmt.Errorf("kubernetes interface cannot be nil")
+	}
+
+	if len(endpoints) == 0 {
```

Comment:
could we move it to validateEndpointConfig? Or maybe is better to keep it with other checks